lib/k8s/util.rb

```diff
@@ -35,7 +38,7 @@ def self.json_patch(a, b)
         operator = diff[0]
         # substitute '/' with '~1' and '~' with '~0'
         # according to RFC 6901
-        path = diff[1].map {|p| p.to_s.gsub('/', '~1')}.map {|p| p.to_s.gsub('~', '~0')}
+        path = diff[1].map {|p| p.to_s.gsub(PATH_REGEX, PATH_TR_MAP) }
```

Comment:
From above:
```
# Each diff is like ["+", "spec.selector.aziz", "kebab"]
# or ["-", "spec.selector.aziz", "kebab"]
```

If this is really the case, the `diff[1].map` can't be right. Also, why the `to_s`, can it be nil? should it stay nil?

Reply:
the comment is outdated, the path in the diff[2] is an array of the path elements. `to_s` as the path elements can be symbols